tripy/scripts/generate_tripy_integration_ir.py

```diff
@@ -0,0 +1,123 @@
+import re
+import shutil
+
+def construct_test_names(output):
+    relevant_output = output.split('\n\n=======================')[0]
+    module_match = re.search(r'<Module (.*?)>', relevant_output)
+    module_name = module_match.group(1) if module_match else ''
+    class_match = re.search(r'<Class (.*?)>', relevant_output)
+    class_name = class_match.group(1) if class_match else ''
+    function_matches = re.findall(r'<Function (.*?)>', relevant_output)
+    
+    test_names = [
+        f"/tripy/{module_name}::{class_name}::{func}" 
+        for func in function_matches
+    ]
+    return test_names
+
+def run_single_test(test_name):
+    with tempfile.TemporaryDirectory() as test_dir:
+        # Set environment for this test
+        test_env = os.environ.copy()
+        test_env['TRIPY_MLIR_DEBUG_ENABLED'] = '1'
+        test_env['TRIPY_MLIR_DEBUG_PATH'] = test_dir
+        
+        # Run the test
+        cmd = ['pytest', test_name, '-v']
+        result = subprocess.run(cmd, env=test_env, capture_output=True, text=True)
+        
+        # Look for 0_ file and get content
+        mlir_files = glob.glob(os.path.join(test_dir, '**', '0_*.mlir'), recursive=True)
+        content = ""
+        
+        if mlir_files:
+            with open(mlir_files[0], 'r') as source:
+                content = source.read()
+                content = re.sub(r'^// -----.*$', '', content, flags=re.MULTILINE)
+                content = content.strip()
+        
+        return result.returncode == 0, content
+
+def collect_and_format_tests(test_pattern='/tripy/tests/integration/*'):
+    # Collect test files
+    test_files = glob.glob(test_pattern)
+    test_assemblies = {}
+    
+    for test_file in test_files:
+        print(f"Collecting tests from {test_file}")
+        cmd = ['pytest', '--collect-only', '--no-header', '--no-summary', test_file]
+        result = subprocess.run(cmd, capture_output=True, text=True)
+        test_names = construct_test_names(result.stdout)
+        
+        # Run tests and collect MLIR
+        print("\nRunning tests and collecting MLIR:")
+        for test_name in test_names:
+            print(f"Running {test_name}")
+            success, content = run_single_test(test_name)
+            status = "PASSED" if success else "FAILED"
+            print(f"{status} {test_name}")
+            
+            if content:
+                test_key = f"test_{test_name.split('::')[-1]}"
+                test_assemblies[test_key] = f"""\n{content}\n"""
+    
+    # Write collected assemblies to Python file
+    with open('test_tripy_integration.py', 'w') as f:
+        f.write("""# RUN: %PYTHON %s | FileCheck %s
+import os
+import mlir_tensorrt.compiler.api as api
+from mlir_tensorrt.compiler.ir import *
+
+# Store assemblies in a dictionary with test names as keys
+TEST_ASSEMBLIES = {
+""")
+        
+        for test_name, asm in test_assemblies.items():
+            f.write(f'    "{test_name}": """{asm}""",\n')
+        
```

Comment:
How large is the generated code usually? Wondering if we might need to split them across tests instead of just test files?

Reply:
I have not generated the code for the complete test suit (WIP). I will update once done.

Reply:
Generated code is about 30k lines and takes around 15 minutes to run end to end in MLIR-TRT. May be we can perform some deduplication i.e. just include first test case in a class. or may be just include a single test per module.

Reply:
Does a naive deduplication help at all? e.g. just comparing the generated StableHLO textual representation directly